File: Python/pe_005.py
#project euler-005

def gcd(a,b):
    a,b=max(a,b),min(a,b)
    while b!=0:
        a,b=b,a%b
    return a

def lcm_all(n):
    lcm=1
    for i in range(1,n+1):
        lcm=lcm*i//gcd(lcm,i)
    return lcm

t=int(input())

for i in range(t):
    n=int(input())
    print(lcm_all(n))

File: Clojure/pe-005.clj
(defn gcd [a b]
    (loop [n1 a n2 b]
        (if (= n2 0)
            n1
            (recur n2 (rem n1 n2)))))

(defn lcm [a b]
    (/ (* a b) (gcd a b)))

(defn smallest-multiple [n]
    (->> (inc n)
         (range 1)
         (reduce lcm)))

(println (smallest-multiple 20))
